src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -118,23 +135,52 @@ export const WorkspacePermissionSettingInput = ({
   );
 
   const handlePermissionModeOptionChange = useCallback(
-    (id: string) => {
-      if (optionIdToWorkspacePermissionModesMap[id]) {
-        onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+    (option: Array<EuiComboBoxOptionOption<any>>) => {
+      if (option.length > 0) {
+        const id = option[0].value;
+        if (optionIdToWorkspacePermissionModesMap[id]) {
+          onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+        }
       }
     },
     [index, onPermissionModesChange]
   );
 
+  const toggleTypeList = () => {
+    if (!isTypeListOpen) {
+      if (type === WorkspacePermissionItemType.User) {
+        setTypeOptions([{ ...typeOptions[0], checked: 'on' }, typeOptions[1]]);
+      } else {
+        setTypeOptions([typeOptions[0], { ...typeOptions[1], checked: 'on' }]);
+      }
+    }
+    setIsTypeListOpen((current) => !current);
+  };
+
+  const handleTypeChange = useCallback(
+    (options: Array<EuiSelectableOption<any>>) => {
+      setTypeOptions(options);
```

Comment:
The `options` is the current selected option(if I understood correctly), but now you set the EuiSelectable options to the current selected value, doesn't sound correct to me. Please correct me if I'm wrong.

Reply:
'options' contains all the options, but some options will have the checked='on' property added or removed